Updates for `Sumida` protocol upgrade (#147)
autonity/docs.autonity.org: validators/register-vali/index.md

To register a validator you need:

- A [running instance of the Autonity Go Client](/node-operators/) running on your host machine, with [networking](/node-operators/install-aut/#network) configured to allow incoming traffic on the following ports:

  - TCP, UDP `30303` for node p2p (DEVp2p) communication for transaction gossiping.
  - TCP `20203` for node p2p (DEVp2p) communication for consensus gossiping.

>   - TCP `8546` for WebSocket RPC connections to the node by your oracle server.

Reviewer: This is dangerous as it seems you need to allow incoming traffic on your rpc port. That is very bad for security. Since the oracle is running on the same host, the port doesn't have to be open to internet traffic.

Then if a validator wants to run the oracle on a different machine, he is allowed to open his rpc port to do so. But it should be done properly and I don't think we should advertise it as default.

Author: Which one?

I checked 20203  with @vpiyush 

The oracle may be on a different host if you mean 8546.

Is this just an edit to that entry "TCP `8546` for WebSocket RPC connections to the node by your oracle server." to qualify as only needed if running on a different host?

Or is the underlying problem that the docs are not clear enough on hosting advice for AGC and AOS and if they should or not be run on the same or different host?

Reviewer: > Which one?

8546

> Is this just an edit to that entry "TCP 8546 for WebSocket RPC connections to the node by your oracle server." to qualify as only needed if running on a different host?

This should definitely be added.

> Or is the underlying problem that the docs are not clear enough on hosting advice for AGC and AOS and if they should or not be run on the same or different host?

This is not a problem imo, it is an operator choice and for now we can leave this kind of advice out of the docs.

Anyways my point is that opening the RPC port of your validator to the internet can be very dangerous if not done properly (there are tools out there which scan the network for open RPC port and see if they can exploit them, for example sending txs using your keys if you left enabled the `personal` namespace. e.g. https://github.com/Consensys/Legions). Therefore:
1. It should only be done if the oracle is on another machine (which I suspect will not be the case for the majority of participants)
2. If you want to do it, you have to be VERY careful about how you do it (e.g. make sure that only the oracle can access your RPC and not other malicious entities. On how to do that I am not sure, maybe @raj-shekhar1 is the one who can know this. I guess maybe firewall rules?). But anyways I think this should translate to a BIG warning on the docs saying something like "If you open your RPC port, make sure that you take appropriate measures to secure it properly".

Reviewer: @cmjc article that explains quite clearly what can happen when leaving an rpc port open to the internet and proposes differnet way to secure the rpc port https://medium.com/coinmonks/securing-your-ethereum-nodes-from-hackers-8b7d5bac8986 and https://blog.ethereum.org/2015/08/29/security-alert-insecurely-configured-geth-can-make-funds-remotely-accessible

Author: OK, so, on the Guide **Install Autonity, Requirements, Networking**, I have edited the `8546` entry for ports the operator may choose to allow incoming traffic on to:

>* `TCP 8546` to make WebSocket RPC connections to the node (for example, if you are operating as a validator node to allow a connection from your oracle server if it is running on a different host).

And, I have put the note on security into a warning box so:

```
::: {.callout-warning title="Securing ports allowing incoming traffic" collapse="false"}
It is assumed that most administrators will want to restrict access to these ports (unless they wish to support public access to the RPC calls for anonymous clients).  However, you should ensure that you can connect to one of the RPC ports from your _local_ machine (e.g. with IP whitelisting in your firewall rules, SSH port forwarding or other technique).
:::
```

Author: I have edited the **Guide, Register as a validator, Prerequisites** to remove it as a rquird open port, moved it to a qualified note, and it now reads:

>To register a validator you need:
>
>- A [running instance of the Autonity Go Client](/node-operators/) running on your host machine, with [networking](/node-operators/install-aut/#network) configured to allow incoming traffic on the following ports:
>
 > - TCP, UDP `30303` for node p2p (DEVp2p) communication for transaction gossiping.
 > - TCP `20203` for node p2p (DEVp2p) communication for consensus gossiping.
>
>::: {.callout-note title="Note" collapse="false"}
>The oracle server providing price data to your validator node must be able to establish a WebSocket RPC connection to your node on TCP port `8546`. If you are running your oracle server on a different host to your AGC, then you will need to allow incoming traffic on that TCP port.
>:::

Author: @vpiyush @lorenzo-dev1 @raj-shekhar1 -  these edits are basically addressing @lorenzo-dev1 concerns - but is it enough?  For example, in the **Guide Install Autonity** - add a section with stated minimum security requirements? e.g.

>::: {.callout-warning title="Security of your node installation" collapse="false"}
As a software system publicly accessible over the Internet, your node installation must be adequately secured!
>
>Standard best practices for securing software allowing incoming traffic from the Internet are assumed. For example, firewalls and the configuration of reverse proxies, as well as additional steps required by your own best practices and security policies, are assumed.
>:::

Author: I've put that draft into **Guide Install Autonity, Requirements** . See commit f1a53d737edb8fe04f6eedccbadd39d12e06b394.

Reviewer: I have added a further small suggestion for the register validator page here https://github.com/autonity/docs.autonity.org/pull/147#discussion_r1502253261

With that addition there should be enough warnings about this that even a newbie validator operator should understand that leaving RPC open is dangerous.

For what concerns me the issue is solved, but I will leave this thread open in case someone else has further comments.

Author: Thanks @lorenzo-dev1 - @vpiyush @raj-shekhar1 - any further comments on this, please?

Author: Resolving - discussed with @vpiyush @lorenzo-dev1 @raj-shekhar1 - who are happy with edits made.

To register a validator you need:

- A [running instance of the Autonity Go Client](/node-operators/) running on your host machine, with [networking](/node-operators/install-aut/#network) configured to allow incoming traffic on the following ports:

  - TCP, UDP `30303` for node p2p (DEVp2p) communication for transaction gossiping.
  - TCP `20203` for node p2p (DEVp2p) communication for consensus gossiping.

::: {.callout-note title="Note" collapse="false"}

> The oracle server providing price data to your validator node must be able to establish a WebSocket RPC connection to your node on TCP port `8546`. If you are running your oracle server on a different host to your AGC, then you will need to allow incoming traffic on that TCP port.

Reviewer: ```suggestion
The oracle server providing price data to your validator node must be able to establish a WebSocket RPC connection to your node on TCP port `8546`. If you are running your oracle server on a different host to your AGC, then you will need to allow incoming traffic on that TCP port. Make sure to do it in a secure manner (use firewall rules, reverse proxies or other security measures to ensure that only your oracle node can connect to the RPC).
```

better safe than sorry

This must be performed on the host machine running the Autonity Go Client, using the `autonity genOwnershipProof` command:

```bash
autonity genOwnershipProof --autonitykeys <AUTONITYKEYS_PATH> --oraclekey <ORACLE_KEY_PATH> <TREASURY_ACCOUNT_ADDRESS>
```

where:

>   - `<AUTONITYKEYS_PATH>`: is the path to the private key file of the node's `autonitykeys` P2P node keys, specified when running the node. (For generating your `autonitykeys` file see the guide [Run Autonity](/node-operators/run-aut/).)

Reviewer: ```suggestion
  - `<AUTONITYKEYS_PATH>`: path of the file containing the node's `autonitykeys`. For generating your `autonitykeys` file see the guide [Run Autonity](/node-operators/run-aut/).
```

```

This signature hex will be required for the registration.

If you are running the Autonity Go Client in a docker container, setup as described in the guide [Run Autonity as a Docker Image](/node-operators/run-aut#run-docker), the proof can be generated as follows. In this example the keys are mounted in a volume simply named `keystore`:

```bash
docker run -t -i --volume $PWD/keystore:/keystore --name autonity --rm ghcr.io/autonity/autonity:latest genOwnershipProof --autonitykeys /<AUTONITYKEYS_PATH> --oraclekey /<ORACLE_KEY_PATH> <TREASURY_ACCOUNT_ADDRESS>
```

::: {.callout-note title="Note" collapse="false"}
The `genOwnershipProof` command options `--autonitykeys` and `--oraclekey` options require the raw (unencrypted) private key file is passed in as argument.

> The `autonitykeys` file is unencrypted.

Reviewer: ```suggestion
The `autonitykeys` file is generated unencrypted.
```

"admin_enode": "enode://cef6334d0855b72dadaa923ceae532550ef68e0ac50288a393eda5d811b9e81053e1324e637a202e21d04e301fe1765900bdd9f3873d58a2badf693331cb1b15@751.11.121.34:30303",
"admin_id": "f8d35fa6019628963668e868a9f070101236476fe077f4a058c0c22e81b8a6c9"
}
```

The url is returned in the `admin_enode` field.

The [validator address](/concepts/validator/#validator-identifier) or [validator identifier](/concepts/validator/#validator-identifier) is derived from the validator [P2P node key](/concepts/validator/#p2p-node-key)'s public key.

When you generated your `autonitykeys` file for the node, the validator address was printed to terminal as the `Node address` as shown in the guide [Run Autonity](/node-operators/run-aut/):

```bash
Node address: 0x550454352B8e1EAD5F27Cce108EF59439B18E249
Node public key: 0xcef6334d0855b72dadaa923ceae532550ef68e0ac50288a393eda5d811b9e81053e1324e637a202e21d04e301fe1765900bdd9f3873d58a2badf693331cb1b15
Consensus public key: 0x90e54b54718c6d5e50d10b93743d743ebcec2f2a2fd43be6813dc5399e11a9bae891c0a357c8f3aa8ca411f9a526a03f
```

::: {.callout-tip title="Tip" collapse="false"}

> Note that the `Node public key` value minus the leading `0x` market of the HEX string is the public key component of your enode url. You can verify you are using the correct `autonitykeys` file by checking the values correspond. If they don't, then you have an `autonitykeys`file mismatch. 

Reviewer: ```suggestion
Note that the `Node public key` value minus the leading `0x` marker of the HEX string is the public key component of your enode url. You can verify you are using the correct `autonitykeys` file by checking the values correspond. If they don't, then you have an `autonitykeys` file mismatch. 
```
